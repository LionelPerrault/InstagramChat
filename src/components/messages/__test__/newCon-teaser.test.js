import React from 'react'
import { create } from 'react-test-renderer'
import NewConTeaser from '../newCon-teaser'
import { shallow } from 'enzyme'

describe('NewConTeaser Component', () => {
  const comp = (
    <NewConTeaser
      userDetails={{
        id: 7,
        username: 'ghalib',
      }}
    />
  )

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should mock message action when button is clicked', () => {
    const wrapper = shallow(comp)
    wrapper.find('SecondaryButton').simulate('click', {
      preventDefault() {},
    })
  })

  it('should redirect to /messages when conversation with user is created', () => {
    const wrapper = shallow(comp)
    wrapper.setState({ messaged: true })
    expect(wrapper.find('Redirect').exists()).toBe(true)
  })
})
