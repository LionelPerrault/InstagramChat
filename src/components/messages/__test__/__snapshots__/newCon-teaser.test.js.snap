// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewConTeaser Component should match snapshot 1`] = `
<div
  className="recomm_teaser"
>
  <span>
    Wanna message 
    ghalib
    ? Create a private conversation with 
    ghalib
    .
  </span>
  <a
    className="sec_btn"
    href="#"
    onClick={[Function]}
  >
    Message
  </a>
</div>
`;
